[nikohomecontrol] Niko home control 2 Digital black thermostat

When adding thermostat in the digital black the binder are crashing

## Expected Behavior
Works as the normal thermostats

## Current Behavior
Throws this exeption:
_2020-07-10 12:10:30.214 [ERROR] [ansport.mqtt.internal.ClientCallback] - MQTT message received. MqttMessageSubscriber#processMessage() implementation failure

java.lang.NullPointerException: null

	at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.updateThermostatState(NikoHomeControlCommunication2.java:496) ~[?:?]

	at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.updateState(NikoHomeControlCommunication2.java:439) ~[?:?]

	at java.util.ArrayList.forEach(ArrayList.java:1257) ~[?:1.8.0_252]

	at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.devicesEvt(NikoHomeControlCommunication2.java:314) ~[?:?]

	at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.processMessage(NikoHomeControlCommunication2.java:778) ~[?:?]

	at org.eclipse.smarthome.io.transport.mqtt.internal.ClientCallback.lambda$3(ClientCallback.java:96) ~[?:?]

	at java.util.ArrayList.forEach(ArrayList.java:1257) ~[?:1.8.0_252]

	at org.eclipse.smarthome.io.transport.mqtt.internal.ClientCallback.messageArrived(ClientCallback.java:96) ~[?:?]

	at org.eclipse.smarthome.io.transport.mqtt.internal.ClientCallback.messageArrived(ClientCallback.java:71) ~[?:?]

	at com.hivemq.client.internal.mqtt.mqtt3.Mqtt3AsyncClientView.lambda$callbackView$1(Mqtt3AsyncClientView.java:73) ~[bundleFile:?]

	at com.hivemq.client.internal.mqtt.MqttAsyncClient$CallbackSubscriber.onNext(MqttAsyncClient.java:227) [bundleFile:?]

	at com.hivemq.client.internal.mqtt.MqttAsyncClient$CallbackSubscriber.onNext(MqttAsyncClient.java:212) [bundleFile:?]

	at com.hivemq.client.rx.FlowableWithSingle$SingleFutureSubscriber.onNext(FlowableWithSingle.java:377) [bundleFile:?]

	at com.hivemq.client.internal.rx.operators.FlowableWithSingleCombine$SplitSubscriber$Default.tryOnNextActual(FlowableWithSingleCombine.java:206) [bundleFile:?]

	at com.hivemq.client.internal.rx.operators.FlowableWithSingleCombine$SplitSubscriber.tryOnNext(FlowableWithSingleCombine.java:171) [bundleFile:?]

	at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnConditionalSubscriber.runAsync(FlowableObserveOn.java:649) [bundleFile:?]

	at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176) [bundleFile:?]

	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66) [bundleFile:?]

	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57) [bundleFile:?]

	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_252]

	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_252]

	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_252]

	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]

	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]

	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]_

## Possible Solution
Implement error handleing for this error

## Steps to Reproduce (for Bugs)
Update digital black to 2.8.1 and add a thermostat function

## Context
Not able to get any informations from the API

## Your Environment
Running on an raspbian

Is this helpful?

> 2020-07-30 15:16:16.440 [TRACE] [l.nhc2.NikoHomeControlCommunication2] - Niko Home Control: received topic hobby/control/devices/evt, payload {"Method":"devices.changed","Params":[{"Devices":[{"PropertyDefinitions":[{"OverruleActive":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"SetpointTemperature":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"false"}},{"OverruleSetpoint":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"true"}},{"OverruleTime":{"Description":"Range(1.00,1439.00,1.00)","HasStatus":"true","CanControl":"true"}},{"Program":{"Description":"Choice(Day,Night,Eco,Off,Cool,Prog1,Prog2,Prog3)","HasStatus":"true","CanControl":"true"}},{"EcoSave":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"Demand":{"Description":"Choice(Heating,Cooling,None)","HasStatus":"true","CanControl":"false"}},{"AmbientTemperature":{"Description":"Range(-5.00,45.00,0.50)","HasStatus":"true","CanControl":"false"}}],"Uuid":"2a6fc15b-9691-4cb8-b606-395c68cd094c"},{"PropertyDefinitions":[{"OverruleActive":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"SetpointTemperature":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"false"}},{"OverruleSetpoint":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"true"}},{"OverruleTime":{"Description":"Range(1.00,1439.00,1.00)","HasStatus":"true","CanControl":"true"}},{"Program":{"Description":"Choice(Day,Night,Eco,Off,Cool,Prog1,Prog2,Prog3)","HasStatus":"true","CanControl":"true"}},{"EcoSave":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"Demand":{"Description":"Choice(Heating,Cooling,None)","HasStatus":"true","CanControl":"false"}},{"AmbientTemperature":{"Description":"Range(-5.00,45.00,0.50)","HasStatus":"true","CanControl":"false"}}],"Uuid":"8f357a31-c840-4210-ae3e-cfac085f6f58"},{"PropertyDefinitions":[{"OverruleActive":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"SetpointTemperature":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"false"}},{"OverruleSetpoint":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"true"}},{"OverruleTime":{"Description":"Range(1.00,1439.00,1.00)","HasStatus":"true","CanControl":"true"}},{"Program":{"Description":"Choice(Day,Night,Eco,Off,Cool,Prog1,Prog2,Prog3)","HasStatus":"true","CanControl":"true"}},{"EcoSave":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"Demand":{"Description":"Choice(Heating,Cooling,None)","HasStatus":"true","CanControl":"false"}},{"AmbientTemperature":{"Description":"Range(-5.00,45.00,0.50)","HasStatus":"true","CanControl":"false"}}],"Uuid":"a294f613-176d-4753-bb8a-0d44247cf4c2"},{"PropertyDefinitions":[{"OverruleActive":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"SetpointTemperature":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"false"}},{"OverruleSetpoint":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"true"}},{"OverruleTime":{"Description":"Range(1.00,1439.00,1.00)","HasStatus":"true","CanControl":"true"}},{"Program":{"Description":"Choice(Day,Night,Eco,Off,Cool,Prog1,Prog2,Prog3)","HasStatus":"true","CanControl":"true"}},{"EcoSave":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"Demand":{"Description":"Choice(Heating,Cooling,None)","HasStatus":"true","CanControl":"false"}},{"AmbientTemperature":{"Description":"Range(-5.00,45.00,0.50)","HasStatus":"true","CanControl":"false"}}],"Uuid":"1d45d223-7afd-41eb-9c76-e94f35b54146"},{"PropertyDefinitions":[{"OverruleActive":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"SetpointTemperature":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"false"}},{"OverruleSetpoint":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"true"}},{"OverruleTime":{"Description":"Range(1.00,1439.00,1.00)","HasStatus":"true","CanControl":"true"}},{"Program":{"Description":"Choice(Day,Night,Eco,Off,Cool,Prog1,Prog2,Prog3)","HasStatus":"true","CanControl":"true"}},{"EcoSave":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"Demand":{"Description":"Choice(Heating,Cooling,None)","HasStatus":"true","CanControl":"false"}},{"AmbientTemperature":{"Description":"Range(-5.00,45.00,0.50)","HasStatus":"true","CanControl":"false"}}],"Uuid":"401a381c-3720-4b93-a7ec-515f4a667d0a"}]}]}
2020-07-30 15:16:16.441 [ERROR] [ansport.mqtt.internal.ClientCallback] - MQTT message received. MqttMessageSubscriber#processMessage() implementation failure
java.lang.NullPointerException: null
        at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.updateThermostatState(NikoHomeControlCommunication2.java:496) ~[?:?]
        at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.updateState(NikoHomeControlCommunication2.java:439) ~[?:?]
        at java.util.ArrayList.forEach(ArrayList.java:1257) ~[?:1.8.0_251]
        at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.devicesEvt(NikoHomeControlCommunication2.java:314) ~[?:?]
        at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.processMessage(NikoHomeControlCommunication2.java:778) ~[?:?]
        at org.eclipse.smarthome.io.transport.mqtt.internal.ClientCallback.lambda$3(ClientCallback.java:96) ~[?:?]
        at java.util.ArrayList.forEach(ArrayList.java:1257) ~[?:1.8.0_251]
        at org.eclipse.smarthome.io.transport.mqtt.internal.ClientCallback.messageArrived(ClientCallback.java:96) ~[?:?]
        at org.eclipse.smarthome.io.transport.mqtt.internal.ClientCallback.messageArrived(ClientCallback.java:71) ~[?:?]
        at com.hivemq.client.internal.mqtt.mqtt3.Mqtt3AsyncClientView.lambda$callbackView$1(Mqtt3AsyncClientView.java:73) ~[bundleFile:?]
        at com.hivemq.client.internal.mqtt.MqttAsyncClient$CallbackSubscriber.onNext(MqttAsyncClient.java:227) [bundleFile:?]
        at com.hivemq.client.internal.mqtt.MqttAsyncClient$CallbackSubscriber.onNext(MqttAsyncClient.java:212) [bundleFile:?]
        at com.hivemq.client.rx.FlowableWithSingle$SingleFutureSubscriber.onNext(FlowableWithSingle.java:377) [bundleFile:?]
        at com.hivemq.client.internal.rx.operators.FlowableWithSingleCombine$SplitSubscriber$Default.tryOnNextActual(FlowableWithSingleCombine.java:206) [bundleFile:?]
        at com.hivemq.client.internal.rx.operators.FlowableWithSingleCombine$SplitSubscriber.tryOnNext(FlowableWithSingleCombine.java:171) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnConditionalSubscriber.runAsync(FlowableObserveOn.java:649) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176) [bundleFile:?]
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66) [bundleFile:?]
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_251]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_251]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_251]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_251]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_251]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_251]

I have been able to catch the following message before the mentioned exception. Is this helpful? If not, please let me know how I can provide more helpful information.

> 2020-07-30 15:16:16.440 [TRACE] [l.nhc2.NikoHomeControlCommunication2] - Niko Home Control: received topic hobby/control/devices/evt, payload {"Method":"devices.changed","Params":[{"Devices":[{"PropertyDefinitions":[{"OverruleActive":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"SetpointTemperature":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"false"}},{"OverruleSetpoint":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"true"}},{"OverruleTime":{"Description":"Range(1.00,1439.00,1.00)","HasStatus":"true","CanControl":"true"}},{"Program":{"Description":"Choice(Day,Night,Eco,Off,Cool,Prog1,Prog2,Prog3)","HasStatus":"true","CanControl":"true"}},{"EcoSave":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"Demand":{"Description":"Choice(Heating,Cooling,None)","HasStatus":"true","CanControl":"false"}},{"AmbientTemperature":{"Description":"Range(-5.00,45.00,0.50)","HasStatus":"true","CanControl":"false"}}],"Uuid":"2a6fc15b-9691-4cb8-b606-395c68cd094c"},{"PropertyDefinitions":[{"OverruleActive":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"SetpointTemperature":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"false"}},{"OverruleSetpoint":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"true"}},{"OverruleTime":{"Description":"Range(1.00,1439.00,1.00)","HasStatus":"true","CanControl":"true"}},{"Program":{"Description":"Choice(Day,Night,Eco,Off,Cool,Prog1,Prog2,Prog3)","HasStatus":"true","CanControl":"true"}},{"EcoSave":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"Demand":{"Description":"Choice(Heating,Cooling,None)","HasStatus":"true","CanControl":"false"}},{"AmbientTemperature":{"Description":"Range(-5.00,45.00,0.50)","HasStatus":"true","CanControl":"false"}}],"Uuid":"8f357a31-c840-4210-ae3e-cfac085f6f58"},{"PropertyDefinitions":[{"OverruleActive":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"SetpointTemperature":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"false"}},{"OverruleSetpoint":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"true"}},{"OverruleTime":{"Description":"Range(1.00,1439.00,1.00)","HasStatus":"true","CanControl":"true"}},{"Program":{"Description":"Choice(Day,Night,Eco,Off,Cool,Prog1,Prog2,Prog3)","HasStatus":"true","CanControl":"true"}},{"EcoSave":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"Demand":{"Description":"Choice(Heating,Cooling,None)","HasStatus":"true","CanControl":"false"}},{"AmbientTemperature":{"Description":"Range(-5.00,45.00,0.50)","HasStatus":"true","CanControl":"false"}}],"Uuid":"a294f613-176d-4753-bb8a-0d44247cf4c2"},{"PropertyDefinitions":[{"OverruleActive":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"SetpointTemperature":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"false"}},{"OverruleSetpoint":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"true"}},{"OverruleTime":{"Description":"Range(1.00,1439.00,1.00)","HasStatus":"true","CanControl":"true"}},{"Program":{"Description":"Choice(Day,Night,Eco,Off,Cool,Prog1,Prog2,Prog3)","HasStatus":"true","CanControl":"true"}},{"EcoSave":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"Demand":{"Description":"Choice(Heating,Cooling,None)","HasStatus":"true","CanControl":"false"}},{"AmbientTemperature":{"Description":"Range(-5.00,45.00,0.50)","HasStatus":"true","CanControl":"false"}}],"Uuid":"1d45d223-7afd-41eb-9c76-e94f35b54146"},{"PropertyDefinitions":[{"OverruleActive":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"SetpointTemperature":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"false"}},{"OverruleSetpoint":{"Description":"Range(5.00,30.00,0.50)","HasStatus":"true","CanControl":"true"}},{"OverruleTime":{"Description":"Range(1.00,1439.00,1.00)","HasStatus":"true","CanControl":"true"}},{"Program":{"Description":"Choice(Day,Night,Eco,Off,Cool,Prog1,Prog2,Prog3)","HasStatus":"true","CanControl":"true"}},{"EcoSave":{"Description":"Boolean","HasStatus":"true","CanControl":"true"}},{"Demand":{"Description":"Choice(Heating,Cooling,None)","HasStatus":"true","CanControl":"false"}},{"AmbientTemperature":{"Description":"Range(-5.00,45.00,0.50)","HasStatus":"true","CanControl":"false"}}],"Uuid":"401a381c-3720-4b93-a7ec-515f4a667d0a"}]}]}
2020-07-30 15:16:16.441 [ERROR] [ansport.mqtt.internal.ClientCallback] - MQTT message received. MqttMessageSubscriber#processMessage() implementation failure
java.lang.NullPointerException: null
        at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.updateThermostatState(NikoHomeControlCommunication2.java:496) ~[?:?]
        at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.updateState(NikoHomeControlCommunication2.java:439) ~[?:?]
        at java.util.ArrayList.forEach(ArrayList.java:1257) ~[?:1.8.0_251]
        at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.devicesEvt(NikoHomeControlCommunication2.java:314) ~[?:?]
        at org.openhab.binding.nikohomecontrol.internal.protocol.nhc2.NikoHomeControlCommunication2.processMessage(NikoHomeControlCommunication2.java:778) ~[?:?]
        at org.eclipse.smarthome.io.transport.mqtt.internal.ClientCallback.lambda$3(ClientCallback.java:96) ~[?:?]
        at java.util.ArrayList.forEach(ArrayList.java:1257) ~[?:1.8.0_251]
        at org.eclipse.smarthome.io.transport.mqtt.internal.ClientCallback.messageArrived(ClientCallback.java:96) ~[?:?]
        at org.eclipse.smarthome.io.transport.mqtt.internal.ClientCallback.messageArrived(ClientCallback.java:71) ~[?:?]
        at com.hivemq.client.internal.mqtt.mqtt3.Mqtt3AsyncClientView.lambda$callbackView$1(Mqtt3AsyncClientView.java:73) ~[bundleFile:?]
        at com.hivemq.client.internal.mqtt.MqttAsyncClient$CallbackSubscriber.onNext(MqttAsyncClient.java:227) [bundleFile:?]
        at com.hivemq.client.internal.mqtt.MqttAsyncClient$CallbackSubscriber.onNext(MqttAsyncClient.java:212) [bundleFile:?]
        at com.hivemq.client.rx.FlowableWithSingle$SingleFutureSubscriber.onNext(FlowableWithSingle.java:377) [bundleFile:?]
        at com.hivemq.client.internal.rx.operators.FlowableWithSingleCombine$SplitSubscriber$Default.tryOnNextActual(FlowableWithSingleCombine.java:206) [bundleFile:?]
        at com.hivemq.client.internal.rx.operators.FlowableWithSingleCombine$SplitSubscriber.tryOnNext(FlowableWithSingleCombine.java:171) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnConditionalSubscriber.runAsync(FlowableObserveOn.java:649) [bundleFile:?]
        at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176) [bundleFile:?]
        at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66) [bundleFile:?]
        at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_251]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_251]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_251]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_251]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_251]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_251]

--- a/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/handler/NikoHomeControlThermostatHandler.java
+++ b/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/handler/NikoHomeControlThermostatHandler.java
@@ -206,7 +206,7 @@ public class NikoHomeControlThermostatHandler extends BaseThingHandler implement
     public void thermostatEvent(int measured, int setpoint, int mode, int overrule, int demand) {
         updateState(CHANNEL_MEASURED, new QuantityType<>(nhcThermostat.getMeasured() / 10.0, CELSIUS));
 
-        long overruletime = nhcThermostat.getRemainingOverruletime();
+        int overruletime = nhcThermostat.getRemainingOverruletime();
         updateState(CHANNEL_OVERRULETIME, new DecimalType(overruletime));
         // refresh the remaining time every minute
         scheduleRefreshOverruletime(nhcThermostat);
@@ -240,7 +240,7 @@ public class NikoHomeControlThermostatHandler extends BaseThingHandler implement
         }
 
         refreshTimer = scheduler.scheduleWithFixedDelay(() -> {
-            long remainingTime = nhcThermostat.getRemainingOverruletime();
+            int remainingTime = nhcThermostat.getRemainingOverruletime();
             updateState(CHANNEL_OVERRULETIME, new DecimalType(remainingTime));
             if (remainingTime <= 0) {
                 cancelRefreshTimer();

--- a/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/NhcThermostat.java
+++ b/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/NhcThermostat.java
@@ -294,11 +294,12 @@ public abstract class NhcThermostat {
     /**
      * @return remaining overrule time in minutes
      */
-    public long getRemainingOverruletime() {
+    public int getRemainingOverruletime() {
         if (overruleStart == null) {
             return 0;
         } else {
-            return overruletime - ChronoUnit.MINUTES.between(overruleStart, LocalDateTime.now());
+            // overruletime time max 23h59min, therefore can safely cast to int
+            return overruletime - (int) ChronoUnit.MINUTES.between(overruleStart, LocalDateTime.now());
         }
     }
 

--- a/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NhcDevice2.java
+++ b/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NhcDevice2.java
@@ -14,72 +14,113 @@ package org.openhab.binding.nikohomecontrol.internal.protocol.nhc2;
 
 import java.util.List;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
 /**
  * {@link NhcDevice2} represents a Niko Home Control II device. It is used when parsing the device response json and
  * when creating the state update json to send to the Connected Controller.
  *
  * @author Mark Herwege - Initial Contribution
  */
+@NonNullByDefault
 class NhcDevice2 {
     static class NhcProperty {
         // fields for lights
+        @Nullable
         String status;
+        @Nullable
         String brightness;
+        @Nullable
         String aligned;
+        @Nullable
         String basicState;
         // fields for motors
+        @Nullable
         String action;
+        @Nullable
         String position;
+        @Nullable
         String moving;
         // fields for thermostats and hvac
+        @Nullable
         String setpointTemperature;
+        @Nullable
         String program;
+        @Nullable
         String overruleActive;
+        @Nullable
         String overruleSetpoint;
+        @Nullable
         String overruleTime;
+        @Nullable
         String ecoSave;
+        @Nullable
         String demand;
+        @Nullable
         String operationMode;
+        @Nullable
         String ambientTemperature;
+        @Nullable
         String protectMode;
+        @Nullable
         String thermostatOn;
+        @Nullable
         String hvacOn;
         // fields for fans and ventilation
+        @Nullable
         String fanSpeed;
         // fields for electricity metering
+        @Nullable
         String electricalEnergy;
+        @Nullable
         String electricalPower;
+        @Nullable
         String reportInstantUsage;
         // fields for access control
+        @Nullable
         String doorlock;
     }
 
     static class NhcTrait {
+        @Nullable
         String macAddress;
         // fields for energyMeters metering
+        @Nullable
         String channel;
+        @Nullable
         String meterType;
     }
 
     static class NhcParameter {
+        @Nullable
         String locationId;
+        @Nullable
         String locationName;
+        @Nullable
         String locationIcon;
         // fields for electricity metering
+        @Nullable
         String flow;
+        @Nullable
         String segment;
+        @Nullable
         String clampType;
+        @Nullable
         String shortName;
     }
 
-    String name;
-    String uuid;
-    String technology;
-    String identifier;
-    String model;
-    String type;
-    String online;
+    String name = "";
+    String uuid = "";
+    String technology = "";
+    String identifier = "";
+    String model = "";
+    String type = "";
+    String online = "";
+    @Nullable
     List<NhcProperty> properties;
+    @Nullable
     List<NhcTrait> traits;
+    @Nullable
     List<NhcParameter> parameters;
 }

--- a/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NhcService2.java
+++ b/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NhcService2.java
@@ -12,13 +12,16 @@
  */
 package org.openhab.binding.nikohomecontrol.internal.protocol.nhc2;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
 /**
  * {@link NhcService2} represents a Niko Home Control II service. It is used when parsing the service response json.
  *
  * @author Mark Herwege - Initial Contribution
  */
+@NonNullByDefault
 class NhcService2 {
-    String name;
+    String name = "";
 
     String name() {
         return name;

--- a/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NikoHomeControlCommunication2.java
+++ b/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NikoHomeControlCommunication2.java
@@ -433,6 +433,11 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
     }
 
     private void updateState(NhcDevice2 device) {
+        if (device.properties == null) {
+            logger.debug("Cannot Update state for {} as no properties defined in device message", device.uuid);
+            return;
+        }
+
         if (actions.containsKey(device.uuid)) {
             updateActionState((NhcAction2) actions.get(device.uuid), device);
         } else if (thermostats.containsKey(device.uuid)) {
@@ -482,14 +487,14 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
     }
 
     private void updateRollershutterState(NhcAction2 action, NhcDevice2 device) {
-        Optional<NhcProperty> positionProperty = device.properties.stream().filter(p -> (p.position != null))
-                .findFirst();
-
-        if (positionProperty.isPresent()) {
-            action.setState(Integer.parseInt(positionProperty.get().position));
-            logger.debug("Niko Home Control: setting action {} internally to {}", action.getId(),
-                    positionProperty.get().position);
-        }
+        device.properties.stream().map(p -> p.position).filter(Objects::nonNull).findFirst().ifPresent(position -> {
+            try {
+                action.setState(Integer.parseInt(position));
+                logger.debug("Niko Home Control: setting action {} internally to {}", action.getId(), position);
+            } catch (NumberFormatException e) {
+                logger.trace("Niko Home Control: received empty rollershutter {} position info", action.getId());
+            }
+        });
     }
 
     private void updateThermostatState(NhcThermostat2 thermostat, NhcDevice2 device) {
@@ -499,53 +504,49 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
                 .filter(Objects::nonNull).map(t -> Math.round(Float.parseFloat(t) * 10)).findFirst();
         Optional<Integer> overruleTimeProperty = device.properties.stream().map(p -> p.overruleTime)
                 .filter(Objects::nonNull).map(t -> Math.round(Float.parseFloat(t))).findFirst();
-        Optional<String> programProperty = device.properties.stream().map(p -> p.program).filter(Objects::nonNull)
-                .findFirst();
         Optional<Integer> setpointTemperatureProperty = device.properties.stream().map(p -> p.setpointTemperature)
                 .filter(Objects::nonNull).map(t -> Math.round(Float.parseFloat(t) * 10)).findFirst();
-        Optional<NhcProperty> ecoSaveProperty = device.properties.stream().filter(p -> (p.ecoSave != null)).findFirst();
+        Optional<Boolean> ecoSaveProperty = device.properties.stream().map(p -> p.ecoSave).filter(Objects::nonNull)
+                .map(t -> Boolean.parseBoolean(t)).findFirst();
         Optional<Integer> ambientTemperatureProperty = device.properties.stream().map(p -> p.ambientTemperature)
                 .filter(Objects::nonNull).map(t -> Math.round(Float.parseFloat(t) * 10)).findFirst();
+        Optional<String> demandProperty = device.properties.stream().map(p -> p.demand).filter(Objects::nonNull)
+                .findFirst();
+        Optional<String> operationModeProperty = device.properties.stream().map(p -> p.operationMode)
+                .filter(Objects::nonNull).findFirst();
 
-        Optional<NhcProperty> demandProperty = device.properties.stream()
-                .filter(p -> ((p.demand != null) || (p.operationMode != null))).findFirst();
-
-        String modeString = programProperty.orElse("");
+        String modeString = device.properties.stream().map(p -> p.program).filter(Objects::nonNull).findFirst()
+                .orElse("");
         int mode = IntStream.range(0, THERMOSTATMODES.length).filter(i -> THERMOSTATMODES[i].equals(modeString))
                 .findFirst().orElse(thermostat.getMode());
 
         int measured = ambientTemperatureProperty.orElse(thermostat.getMeasured());
         int setpoint = setpointTemperatureProperty.orElse(thermostat.getSetpoint());
 
-        int overrule = 0;
-        int overruletime = 0;
+        int overrule = thermostat.getOverrule();
+        int overruletime = thermostat.getRemainingOverruletime();
         if (overruleActiveProperty.orElse(false)) {
             overrule = overruleSetpointProperty.orElse(0);
             overruletime = overruleTimeProperty.orElse(0);
         }
 
-        int ecosave = ecoSaveProperty.isPresent() ? ("True".equals(ecoSaveProperty.get().ecoSave) ? 1 : 0)
-                : thermostat.getEcosave();
+        int ecosave = thermostat.getEcosave();
+        if (ecoSaveProperty.orElse(false)) {
+            ecosave = 1;
+        }
 
         int demand = thermostat.getDemand();
-        if (demandProperty.isPresent()) {
-            String demandString;
-            if (demandProperty.get().demand != null) {
-                demandString = demandProperty.get().demand;
-            } else {
-                demandString = demandProperty.get().operationMode;
-            }
-            switch (demandString) {
-                case "None":
-                    demand = 0;
-                    break;
-                case "Heating":
-                    demand = 1;
-                    break;
-                case "Cooling":
-                    demand = -1;
-                    break;
-            }
+        String demandString = (demandProperty.orElse(operationModeProperty.orElse("")));
+        switch (demandString) {
+            case "None":
+                demand = 0;
+                break;
+            case "Heating":
+                demand = 1;
+                break;
+            case "Cooling":
+                demand = -1;
+                break;
         }
 
         logger.debug(

--- a/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NikoHomeControlCommunication2.java
+++ b/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NikoHomeControlCommunication2.java
@@ -353,9 +353,11 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
     }
 
     private void addDevice(NhcDevice2 device) {
-        String location;
-        location = device.parameters.stream().map(p -> p.locationName).filter(Objects::nonNull).findFirst()
-                .orElse(null);
+        String location = null;
+        if (device.parameters != null) {
+            location = device.parameters.stream().map(p -> p.locationName).filter(Objects::nonNull).findFirst()
+                    .orElse(null);
+        }
 
         if ("action".equals(device.type)) {
             if (!actions.containsKey(device.uuid)) {
@@ -433,33 +435,34 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
     }
 
     private void updateState(NhcDevice2 device) {
-        if (device.properties == null) {
+        List<NhcProperty> deviceProperties = device.properties;
+        if (deviceProperties == null) {
             logger.debug("Cannot Update state for {} as no properties defined in device message", device.uuid);
             return;
         }
 
         if (actions.containsKey(device.uuid)) {
-            updateActionState((NhcAction2) actions.get(device.uuid), device);
+            updateActionState((NhcAction2) actions.get(device.uuid), deviceProperties);
         } else if (thermostats.containsKey(device.uuid)) {
-            updateThermostatState((NhcThermostat2) thermostats.get(device.uuid), device);
+            updateThermostatState((NhcThermostat2) thermostats.get(device.uuid), deviceProperties);
         } else if (energyMeters.containsKey(device.uuid)) {
-            updateEnergyMeterState((NhcEnergyMeter2) energyMeters.get(device.uuid), device);
+            updateEnergyMeterState((NhcEnergyMeter2) energyMeters.get(device.uuid), deviceProperties);
         }
     }
 
-    private void updateActionState(NhcAction2 action, NhcDevice2 device) {
+    private void updateActionState(NhcAction2 action, List<NhcProperty> deviceProperties) {
         if (action.getType() == ActionType.ROLLERSHUTTER) {
-            updateRollershutterState(action, device);
+            updateRollershutterState(action, deviceProperties);
         } else {
-            updateLightState(action, device);
+            updateLightState(action, deviceProperties);
         }
     }
 
-    private void updateLightState(NhcAction2 action, NhcDevice2 device) {
-        Optional<NhcProperty> statusProperty = device.properties.stream().filter(p -> (p.status != null)).findFirst();
-        Optional<NhcProperty> dimmerProperty = device.properties.stream().filter(p -> (p.brightness != null))
+    private void updateLightState(NhcAction2 action, List<NhcProperty> deviceProperties) {
+        Optional<NhcProperty> statusProperty = deviceProperties.stream().filter(p -> (p.status != null)).findFirst();
+        Optional<NhcProperty> dimmerProperty = deviceProperties.stream().filter(p -> (p.brightness != null))
                 .findFirst();
-        Optional<NhcProperty> basicStateProperty = device.properties.stream().filter(p -> (p.basicState != null))
+        Optional<NhcProperty> basicStateProperty = deviceProperties.stream().filter(p -> (p.basicState != null))
                 .findFirst();
 
         String booleanState = null;
@@ -486,8 +489,8 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
         }
     }
 
-    private void updateRollershutterState(NhcAction2 action, NhcDevice2 device) {
-        device.properties.stream().map(p -> p.position).filter(Objects::nonNull).findFirst().ifPresent(position -> {
+    private void updateRollershutterState(NhcAction2 action, List<NhcProperty> deviceProperties) {
+        deviceProperties.stream().map(p -> p.position).filter(Objects::nonNull).findFirst().ifPresent(position -> {
             try {
                 action.setState(Integer.parseInt(position));
                 logger.debug("Niko Home Control: setting action {} internally to {}", action.getId(), position);
@@ -497,25 +500,25 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
         });
     }
 
-    private void updateThermostatState(NhcThermostat2 thermostat, NhcDevice2 device) {
-        Optional<Boolean> overruleActiveProperty = device.properties.stream().map(p -> p.overruleActive)
+    private void updateThermostatState(NhcThermostat2 thermostat, List<NhcProperty> deviceProperties) {
+        Optional<Boolean> overruleActiveProperty = deviceProperties.stream().map(p -> p.overruleActive)
                 .filter(Objects::nonNull).map(t -> Boolean.parseBoolean(t)).findFirst();
-        Optional<Integer> overruleSetpointProperty = device.properties.stream().map(p -> p.overruleSetpoint)
+        Optional<Integer> overruleSetpointProperty = deviceProperties.stream().map(p -> p.overruleSetpoint)
                 .filter(Objects::nonNull).map(t -> Math.round(Float.parseFloat(t) * 10)).findFirst();
-        Optional<Integer> overruleTimeProperty = device.properties.stream().map(p -> p.overruleTime)
+        Optional<Integer> overruleTimeProperty = deviceProperties.stream().map(p -> p.overruleTime)
                 .filter(Objects::nonNull).map(t -> Math.round(Float.parseFloat(t))).findFirst();
-        Optional<Integer> setpointTemperatureProperty = device.properties.stream().map(p -> p.setpointTemperature)
+        Optional<Integer> setpointTemperatureProperty = deviceProperties.stream().map(p -> p.setpointTemperature)
                 .filter(Objects::nonNull).map(t -> Math.round(Float.parseFloat(t) * 10)).findFirst();
-        Optional<Boolean> ecoSaveProperty = device.properties.stream().map(p -> p.ecoSave).filter(Objects::nonNull)
+        Optional<Boolean> ecoSaveProperty = deviceProperties.stream().map(p -> p.ecoSave).filter(Objects::nonNull)
                 .map(t -> Boolean.parseBoolean(t)).findFirst();
-        Optional<Integer> ambientTemperatureProperty = device.properties.stream().map(p -> p.ambientTemperature)
+        Optional<Integer> ambientTemperatureProperty = deviceProperties.stream().map(p -> p.ambientTemperature)
                 .filter(Objects::nonNull).map(t -> Math.round(Float.parseFloat(t) * 10)).findFirst();
-        Optional<String> demandProperty = device.properties.stream().map(p -> p.demand).filter(Objects::nonNull)
-                .findFirst();
-        Optional<String> operationModeProperty = device.properties.stream().map(p -> p.operationMode)
+        Optional<@Nullable String> demandProperty = deviceProperties.stream().map(p -> p.demand)
+                .filter(Objects::nonNull).findFirst();
+        Optional<@Nullable String> operationModeProperty = deviceProperties.stream().map(p -> p.operationMode)
                 .filter(Objects::nonNull).findFirst();
 
-        String modeString = device.properties.stream().map(p -> p.program).filter(Objects::nonNull).findFirst()
+        String modeString = deviceProperties.stream().map(p -> p.program).filter(Objects::nonNull).findFirst()
                 .orElse("");
         int mode = IntStream.range(0, THERMOSTATMODES.length).filter(i -> THERMOSTATMODES[i].equals(modeString))
                 .findFirst().orElse(thermostat.getMode());
@@ -536,7 +539,8 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
         }
 
         int demand = thermostat.getDemand();
-        String demandString = (demandProperty.orElse(operationModeProperty.orElse("")));
+        String demandString = demandProperty.orElse(operationModeProperty.orElse(""));
+        demandString = demandString == null ? "" : demandString;
         switch (demandString) {
             case "None":
                 demand = 0;
@@ -555,8 +559,8 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
         thermostat.updateState(measured, setpoint, mode, overrule, overruletime, ecosave, demand);
     }
 
-    private void updateEnergyMeterState(NhcEnergyMeter2 energyMeter, NhcDevice2 device) {
-        device.properties.stream().map(p -> p.electricalPower).filter(Objects::nonNull).findFirst()
+    private void updateEnergyMeterState(NhcEnergyMeter2 energyMeter, List<NhcProperty> deviceProperties) {
+        deviceProperties.stream().map(p -> p.electricalPower).filter(Objects::nonNull).findFirst()
                 .ifPresent(electricalPower -> {
                     try {
                         energyMeter.setPower(Integer.parseInt(electricalPower));
@@ -584,9 +588,10 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
         devices.add(device);
         param.devices = devices;
         device.uuid = actionId;
-        device.properties = new ArrayList<>();
+        ArrayList<NhcProperty> deviceProperties = new ArrayList<>();
         NhcProperty property = new NhcProperty();
-        device.properties.add(property);
+        deviceProperties.add(property);
+        device.properties = deviceProperties;
 
         NhcAction2 action = (NhcAction2) actions.get(actionId);
 
@@ -646,15 +651,17 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
         devices.add(device);
         param.devices = devices;
         device.uuid = thermostatId;
-        device.properties = new ArrayList<>();
+        ArrayList<NhcProperty> deviceProperties = new ArrayList<>();
 
         NhcProperty overruleActiveProp = new NhcProperty();
-        device.properties.add(overruleActiveProp);
+        deviceProperties.add(overruleActiveProp);
         overruleActiveProp.overruleActive = "False";
 
         NhcProperty program = new NhcProperty();
-        device.properties.add(program);
+        deviceProperties.add(program);
         program.program = mode;
+
+        device.properties = deviceProperties;
 
         String topic = profile + "/control/devices/cmd";
         String gsonMessage = gson.toJson(message);
@@ -675,25 +682,26 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
         devices.add(device);
         param.devices = devices;
         device.uuid = thermostatId;
-        device.properties = new ArrayList<>();
+        ArrayList<NhcProperty> deviceProperties = new ArrayList<>();
 
         if (overruleTime > 0) {
             NhcProperty overruleActiveProp = new NhcProperty();
             overruleActiveProp.overruleActive = "True";
-            device.properties.add(overruleActiveProp);
+            deviceProperties.add(overruleActiveProp);
 
             NhcProperty overruleSetpointProp = new NhcProperty();
             overruleSetpointProp.overruleSetpoint = String.valueOf(overruleTemp / 10.0);
-            device.properties.add(overruleSetpointProp);
+            deviceProperties.add(overruleSetpointProp);
 
             NhcProperty overruleTimeProp = new NhcProperty();
             overruleTimeProp.overruleTime = String.valueOf(overruleTime);
-            device.properties.add(overruleTimeProp);
+            deviceProperties.add(overruleTimeProp);
         } else {
             NhcProperty overruleActiveProp = new NhcProperty();
             overruleActiveProp.overruleActive = "False";
-            device.properties.add(overruleActiveProp);
+            deviceProperties.add(overruleActiveProp);
         }
+        device.properties = deviceProperties;
 
         String topic = profile + "/control/devices/cmd";
         String gsonMessage = gson.toJson(message);
@@ -714,11 +722,12 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
         devices.add(device);
         param.devices = devices;
         device.uuid = energyMeterId;
-        device.properties = new ArrayList<>();
+        ArrayList<NhcProperty> deviceProperties = new ArrayList<>();
 
         NhcProperty reportInstantUsageProp = new NhcProperty();
-        device.properties.add(reportInstantUsageProp);
+        deviceProperties.add(reportInstantUsageProp);
         reportInstantUsageProp.reportInstantUsage = "True";
+        device.properties = deviceProperties;
 
         String topic = profile + "/control/devices/cmd";
         String gsonMessage = gson.toJson(message);

--- a/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NikoHomeControlCommunication2.java
+++ b/bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NikoHomeControlCommunication2.java
@@ -309,6 +309,9 @@ public class NikoHomeControlCommunication2 extends NikoHomeControlCommunication
             return;
         } else if ("devices.added".equals(method)) {
             deviceList.forEach(this::addDevice);
+        } else if ("devices.changed".contentEquals(method)) {
+            deviceList.forEach(this::removeDevice);
+            deviceList.forEach(this::addDevice);
         }
 
         deviceList.forEach(this::updateState);

--- a/bundles/org.openhab.binding.nikohomecontrol/src/main/resources/ESH-INF/thing/thing-types.xml
+++ b/bundles/org.openhab.binding.nikohomecontrol/src/main/resources/ESH-INF/thing/thing-types.xml
@@ -101,7 +101,7 @@
 		<label>Switch</label>
 		<description>On/Off type action in Niko Home Control</description>
 		<channels>
-			<channel id="switch" typeId="switch"/>
+			<channel id="switch" typeId="system.power"/>
 		</channels>
 		<config-description>
 			<parameter name="actionId" type="text" required="true">
@@ -119,7 +119,7 @@
 		<label>Dimmer</label>
 		<description>Dimmer type action in Niko Home Control</description>
 		<channels>
-			<channel id="brightness" typeId="brightness"/>
+			<channel id="brightness" typeId="system.brightness"/>
 		</channels>
 		<config-description>
 			<parameter name="actionId" type="text" required="true">
@@ -205,18 +205,6 @@
 		<description>Pushbutton control for action in Niko Home Control</description>
 		<category>Switch</category>
 		<autoUpdatePolicy>veto</autoUpdatePolicy>
-	</channel-type>
-	<channel-type id="switch">
-		<item-type>Switch</item-type>
-		<label>Switch</label>
-		<description>Switch control for action in Niko Home Control</description>
-		<category>Switch</category>
-	</channel-type>
-	<channel-type id="brightness">
-		<item-type>Dimmer</item-type>
-		<label>Brightness</label>
-		<description>Brightness control for dimmer action in Niko Home Control</description>
-		<category>DimmableLight</category>
 	</channel-type>
 	<channel-type id="rollershutter">
 		<item-type>Rollershutter</item-type>

